pypowsybl/dynamic.py

```diff
@@ -0,0 +1,190 @@
+
+    def add_generator_synchronous_three_windings_proportional_regulations(self, static_id: str, dynamic_param: str) -> None:
+        self.add_all_dynamic_mappings(static_id=static_id,
+                                      parameter_set_id=dynamic_param,
+                                      mapping_type=DynamicMappingType.GENERATOR_SYNCHRONOUS_THREE_WINDINGS_PROPORTIONAL_REGULATIONS)
+
+    def add_generator_synchronous_four_windings(self, static_id: str, dynamic_param: str) -> None:
+        self.add_all_dynamic_mappings(static_id=static_id,
+                                      parameter_set_id=dynamic_param,
+                                      mapping_type=DynamicMappingType.GENERATOR_SYNCHRONOUS_FOUR_WINDINGS)
+
+    def add_generator_synchronous_four_windings_proportional_regulations(self, static_id: str, dynamic_param: str) -> None:
+        self.add_all_dynamic_mappings(static_id=static_id,
+                                      parameter_set_id=dynamic_param,
+                                      mapping_type=DynamicMappingType.GENERATOR_SYNCHRONOUS_FOUR_WINDINGS_PROPORTIONAL_REGULATIONS)
+
+    def add_current_limit_automaton(self, static_id: str, dynamic_param: str, branch_side: BranchSide) -> None:
+        _pp.add_current_limit_automaton(
+            self._handle, static_id, dynamic_param, branch_side.value)
+
+    def add_all_dynamic_mappings(self, df: _pd.DataFrame = None, **kwargs: _Union[str, DynamicMappingType]) -> None:
+        """
+        Update the dynamic mapping of a simulation, must provide a :class:`~pandas.DataFrame` or as named arguments.
+        The dataframe must contains these three columns :
+            - static_id: id of the network element to map (or the id of the generator for omega_ref)
+            - parameter_set_id: set id in the parameter file
+            - mapping_type: value of enum DynamicMappingType
+        For current_limit_automaton the branch side defaults to ONE, see add_current_limit_automaton to add a model with another branch side.
+        For omega_ref the static_id column refers to the generator's id, and the parameter_set_id will be ignored
+        """
+        metadata = _pp.get_dynamic_mappings_meta_data()
+        if kwargs:
+            kwargs = _add_index_to_kwargs(metadata, **kwargs)
+        df = _adapt_df_or_kwargs(metadata, df, **kwargs)
+        c_df = _create_c_dataframe(df, metadata)
+        _pp.add_all_dynamic_mappings(self._handle, c_df)
+
+
+class CurveMapping:
+    def __init__(self) -> None:
+        self._handle = _pp.create_timeseries_mapping()
+
+    def add_curve(self, dynamic_id: str, variable: str) -> None:
+        _pp.add_curve(self._handle, dynamic_id, variable)
+
+    def add_curves(self, dynamic_id: str, variables: _List[str]) -> None:
+        for var in variables:
+            _pp.add_curve(self._handle, dynamic_id, var)
+
+
+class EventType(_Enum):
+    QUADRIPOLE_DISCONNECT = 'QUADRIPOLE_DISCONNECT'
```

Comment:
We really need to re-work the naming of the events:
- I don't think "quadripole" is an english word ? In any case, in powsybl, we talk about "branch" instead
- I am not sure what "set point boolean" means: does it change the regulation status of a generator ?
- BRANCH_DISCONNECTION : is it a duplicate of QUADRIPOLE_DISCONNECT?  :)

Reply:
From what I understood in powsybl-dynawaltz package, "set point boolean" events can only be used with a generator id. It allows to connect/disconnect the generator.
I'll rename quadripole to branch in pypowsybl, but powsybl-dynawaltz still use "quadripole".
And yes now that the naming is clearer, BRANCH_DISCONNECTION and QUADRIPOLE_DISCONNECT are the same event (and refer to the QUADRIPOLE_DISCONNECT  event in powsybl-dynawaltz)